src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -675,6 +712,285 @@ GroupPrincipal getGroupPrincipal(Member group) {
+              this.soapFactory.newDocumentManagement(authenticationToken);
+          Node node = getNodeById(documentManagement, nodeId);
+          if (node != null) {
+            XMLGregorianCalendar xmlCalendar = node.getModifyDate();
+            if (xmlCalendar != null) {
+              Date date = xmlCalendar.toGregorianCalendar().getTime();
+              this.lastModDocId = nodeId;
+              this.lastModDate = new SimpleDateFormat("yyyyMMdd").format(date);
+              this.lastModTime = new SimpleDateFormat("HHmmss").format(date);
+            }
+          }
+        }
+      }
+    } while (resultCount > 0);
+    log.log(Level.FINER, "Sending modified doc ids: {0}", docIds);
+    pusher.pushDocIds(docIds);
+  }
+
+  /* Reads a SearchResult element from Content Server's XML
+   * search results and builds a DocId by pulling various parts
+   * from the SearchResult, including the names and ids of parent
+   * elements and the result item itself. The DocId has the
+   * form "<start point id>/<name>/<name>/<name>:<object id>".
+   *
+   * An example SearchResult element showing only the elements
+   * used to construct the DocId:
+   *
+   * <SearchResult>
+   *   <OTLocation><![CDATA[2000 1234 5678 -5678 9012]]></OTLocation>
+   *   <OTLocationPath>
+   *     <LocationPathString>
+   *     Enterprise:Folder 1:Project 1
+   *     </LocationPathString>
+   *   </OTLocationPath>
+   *   <OTName>
+   *     Document in Project
+   *     <Value lang="en">Document in Project</Value>
+   *   </OTName>
+   * </SearchResult>
+   */
+  @VisibleForTesting
+  String getXmlSearchDocId(Element resultElement) {
+    // Get the list of ids defining the path to the search result
+    // item from the OTLocation element.
+    List<String> ids = getXmlSearchIds(resultElement);
+    if (ids.size() == 0) {
+      return null;
+    }
+
+    // OTLocation includes the search result item's id as the
+    // final element. The corresponding list of names doesn't
+    // include the search result item, so remove the item id to
+    // make the lists parallel.
+    String objectId = ids.remove(ids.size() - 1);
+
+    // Look through the StartPoint list. If the result item is a
+    // start point, return its id. Otherwise, find the index of a
+    // start point in the id path.
+    StartPoint startPoint = null;
+    int startPointIndex = -1;
+    for (StartPoint sp : this.startPoints) {
+      String startPointId = String.valueOf(sp.getNodeId());
+      if (startPointId.equals(objectId)) {
+        return escapeParam(sp.getName()) + ":" + startPointId;
+      } else {
+        startPointIndex = ids.indexOf(startPointId);
+        if (startPointIndex > -1) {
+          startPoint = sp;
+          break;
+        }
+      }
+    }
+    // If the result item is not contained under a configured
+    // start point, skip it.
+    if (startPointIndex == -1) {
+      log.log(Level.FINE, "No start point in path: " + ids);
```

Comment:
Let's be more clear in the log messages that we are skipping this document/search result. Even just "No start point in path; skipping " + ids and similar below.